From 06ba0f7347c22378ef15733fbf124a649b1c73f0 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Sat, 14 Nov 2020 07:06:24 +0200
Subject: [PATCH] [SW-14663] log: arrange error log when validation fails

make it simple, no negative logic
unify common string to save foot print of ppboot
add new line at the end to turn log clearer

Change-Id: Ia072fc2f6ef59621d4cd97068a5569a1be0860ae
---

diff --git a/ppboot/ppboot.c b/ppboot/ppboot.c
index 4c111b6..b044a80 100644
--- a/ppboot/ppboot.c
+++ b/ppboot/ppboot.c
@@ -60,15 +60,16 @@
 			preboot_puts("Checksum is not valid.\n");
 	}
 	if (!is_valid) {
-		if (!isPri) {
-			bit = CPU_BOOT_ERR0_SEC_IMG_VER_FAIL;
-			msg = "Secondary is not a valid image.";
-		} else {
+		if (isPri) {
 			bit = CPU_BOOT_ERR0_PRI_IMG_VER_FAIL;
-			msg = "Primary is not a valid image.";
+			msg = "Primary";
+		} else {
+			bit = CPU_BOOT_ERR0_SEC_IMG_VER_FAIL;
+			msg = "Secondary";
 		}
 		cpu_boot_err_clrset(bit, true);
 		preboot_puts_raw(msg);
+		preboot_puts(" is not a valid image.\n");
 	}
 
 	return is_valid;
